src/Code.elm

```diff
@@ -0,0 +1,391 @@
+    | DelayForceChar
+      -- ? , ` [ ] @ |
+      -- Currently not all commas in the pretty-print output are marked up as DelimiterChar - we miss
+      -- out characters emitted by Pretty.hs helpers like Pretty.commas.
+    | DelimiterChar
+    | Parenthesis
+    | LinkKeyword -- `typeLink` and `termLink`
+      -- [: :] @[]
+    | DocDelimiter
+      -- the 'include' in @[include], etc
+    | DocKeyword
+
+
+
+-- VIEW
+
+
+syntaxTypeToClassName : SyntaxType -> String
+syntaxTypeToClassName sType =
+    case sType of
+        NumericLiteral ->
+            "numberic-literal"
+
+        TextLiteral ->
+            "text-literal"
+
+        BytesLiteral ->
+            "bytes-literal"
+
+        CharLiteral ->
+            "char-literal"
+
+        BooleanLiteral ->
+            "boolean-literal"
+
+        Blank ->
+            "blank"
+
+        Var ->
+            "var"
+
+        Reference _ ->
+            "reference"
+
+        Referent _ ->
+            "referent"
+
+        Op seqOp ->
+            case seqOp of
+                Cons ->
+                    "op cons"
+
+                Snoc ->
+                    "op snoc"
+
+                Concat ->
+                    "op concat"
+
+        Constructor ->
+            "constructor"
+
+        Request ->
+            "request"
+
+        AbilityBraces ->
+            "ability-braces"
+
+        ControlKeyword ->
+            "control-keyword"
+
+        TypeOperator ->
+            "type-operator"
+
+        BindingEquals ->
+            "binding-equals"
+
+        TypeAscriptionColon ->
+            "type-ascription-colon"
+
+        DataTypeKeyword ->
+            "data-type-keyword"
+
+        DataTypeParams ->
+            "data-type-params"
+
+        Unit ->
+            "unit"
+
+        DataTypeModifier ->
+            "data-type-modifier"
+
+        UseKeyword ->
+            "use-keyword"
+
+        UsePrefix ->
+            "use-prefix"
+
+        UseSuffix ->
+            "use-suffix"
+
+        HashQualifier _ ->
+            "hash-qualifier"
+
+        DelayForceChar ->
+            "delay-force-char"
+
+        DelimiterChar ->
+            "delimeter-char"
+
+        Parenthesis ->
+            "parenthesis"
+
+        LinkKeyword ->
+            "link-keyword"
+
+        DocDelimiter ->
+            "doc-delimeter"
+
+        DocKeyword ->
+            "doc-keyword"
+
+
+viewSegment : SyntaxSegment -> Html msg
+viewSegment (SyntaxSegment sType sText) =
+    span [ class (syntaxTypeToClassName sType) ] [ text sText ]
+
+
+viewSyntaxSegments : NEL.Nonempty SyntaxSegment -> List (Html msg)
+viewSyntaxSegments segments =
+    segments
+        |> NEL.map viewSegment
+        |> NEL.toList
+
+
+view : Code -> Html msg
+view code_ =
+    let
+        content =
+            case code_ of
+                Syntax segments ->
+                    viewSyntaxSegments segments
+
+                Builtin ->
+                    [ span [ class "builtin" ] [ text "Builtin" ] ]
```

Comment:
This is a little incomplete: Builtins should render their type signature.

Reply:
Yes. I have informally been using:

```Haskell 
builtin List.take : Nat -> [a] -> [a]
```

When giving signatures of builtins.

Reply:
Added in https://github.com/unisonweb/codebase-ui/pull/7